tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py

```diff
@@ -632,3 +632,59 @@ def test_setitem_wrong_domain():
 
     with pytest.raises(ValueError, match=r"Incompatible 'Domain'.*"):
         field[(1, slice(None))] = value_incompatible
+
+
```

Comment:
maybe also add a testcase with a 2D connectivity?

Reply:
To be honest I don't really understand if it makes sense to restrict the connectivity in that case. I have to think about it.

Reply:
note to myself:

Start from this case and look at corner cases
```
E_START, E_STOP = 0, 1
V_START, V_STOP = 0, 1

e2v_conn = common._connectivity([0,1], domain=domain(named_range(E,1), named_range(E2V, 2)))
```

the inverse image is probably

```
domain(named_range(E,0), named_range(E2V, 1))
```

Reply:
I've added some cases to test 2d connectivities